legend-engine-xt-nonrelationalStore-mongodb-pure/src/main/resources/core_nonrelational_mongodb/pureToDatabaseCommand/pureToDatabaseCommand.pure

```diff
@@ -84,29 +84,40 @@ function meta::external::store::mongodb::functions::pureToDatabaseCommand::proce
     if ($nextValue->instanceOf(SimpleFunctionExpression),
       |
-        let leftPart = $simpleFunctionExpression.parametersValues->at(0)->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression);
-        let rightPart = $simpleFunctionExpression.func.name->toOne();
-        let currentPath = if ($path == '', | $rightPart, | $rightPart + '.' + $path);
-        buildFilterPath($leftPart, $currentPath);,
+        if ($simpleFunctionExpression.func == toOne_T_MANY__T_1_,
+          | 
+            let functionExpression = $simpleFunctionExpression.parametersValues->at(0)->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression);
+            buildFilterPathRecursively($functionExpression, $path);,
+          |
+            let functionExpression = $simpleFunctionExpression.parametersValues->at(0)->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression);
+            let key = $simpleFunctionExpression.func.name->toOne();
+            let currentPath = buildDotNotationPath($path, $key);
+            buildFilterPathRecursively($functionExpression, $currentPath);
+        );,
       |
-        let leftPart = $simpleFunctionExpression.func.name->toOne();
-        $leftPart + '.' + $path;
+        let key = $simpleFunctionExpression.func.name->toOne();
+        buildDotNotationPath($path, $key);
     );,
   |
-    let rightPart = $simpleFunctionExpression.parametersValues->at(1)->cast(@StoreMappingRoutedValueSpecification).value->cast(@FunctionRoutedValueSpecification).value->cast(@InstanceValue).values->cast(@LambdaFunction<Any>).expressionSequence->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression).func.name->toOne();
-    let leftPart = $simpleFunctionExpression.parametersValues->at(0)->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression);
-    let currentPath = if ($path == '', | $rightPart, | $rightPart + '.' + $path);
-    buildFilterPath($leftPart, $currentPath);
+    let key = $simpleFunctionExpression.parametersValues->at(1)->cast(@StoreMappingRoutedValueSpecification).value->cast(@FunctionRoutedValueSpecification).value->cast(@InstanceValue).values->cast(@LambdaFunction<Any>).expressionSequence->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression).func.name->toOne();
+    let functionExpression = $simpleFunctionExpression.parametersValues->at(0)->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression);
+    let currentPath = buildDotNotationPath($path, $key);
+    buildFilterPathRecursively($functionExpression, $currentPath);
   );
 }
 
+function meta::external::store::mongodb::functions::pureToDatabaseCommand::buildDotNotationPath(currentPath: String[1], pathToAppend: String[1]): String[1]
+{
+  if ($currentPath == '', | $pathToAppend, | $pathToAppend + '.' + $currentPath);
```

Comment:
do we need change the variable names?   We are pre-fixing the "path to append" not appending

Reply:
yes right, this should pathToPrepend

Reply:
makes sense, pathtoPrefix